storey/aggregations.py

```diff
@@ -125,6 +123,11 @@ async def _emit_worker(self):
             await self._emit_all_events(next_emit_time * 1000)
             next_emit_time = next_emit_time + seconds_to_sleep_between_emits
 
+    def run(self):
+        closeables = super().run()
+        closeables.append(self._cache)
+        return closeables
```

Comment:
Probably easier to have each step declare its closables and have run do the append.